### ISSUE TITLE
Installation issue on Mac M1 Max: No matching distribution found for rdkit-pypi==2021.3.5  

### ISSUE THREAD

**Issue by mccarren** [Merck/matcher#7]
I followed all of the instructions and I was having trouble launching the docker image from my Mac M1 Max.  I was wondering if you have seen the error below before.  Another user attempting the process on a PC did not have the same issue.  If I try to run the same conda command from the docker image on my Mac I get the same error.  The docker image should run as Linux OS though.  Sorry- I cannot seem to figure this out.

1.) If you catch the error in the docker-compose console it reads:
 => => # Pip subprocess error:                                                  
 => => # ERROR: Could not find a version that satisfies the requirement rdkit-p 
 => => # ypi==2021.3.5                                                          
 => => # ERROR: No matching distribution found for rdkit-pypi==2021.3.5         
 => => # CondaEnvException: Pip failed     

2.) The remaining output shows below:
docker-compose up    
[+] Building 381.3s (9/12)                                                      
 => [backend internal] load .dockerignore                                  0.0s
 => => transferring context: 2B                                            0.0s
 => [backend internal] load build definition from Dockerfile               0.0s
 => => transferring dockerfile: 490B                                       0.0s
 => [backend internal] load metadata for docker.io/condaforge/miniforge3:  0.6s
 => [backend 1/8] FROM [docker.io/condaforge/miniforge3@sha256:285a0927e33](mailto:docker.io/condaforge/miniforge3@sha256:285a0927e33)  0.0s
 => [backend internal] load build context                                  0.0s
 => => transferring context: 3.52kB                                        0.0s
 => CACHED [backend 2/8] RUN apt-get update -y && apt-get install libxren  0.0s
 => CACHED [backend 3/8] WORKDIR /opt                                      0.0s
 => CACHED [backend 4/8] COPY environment.yml requirements.txt ./          0.0s
 => ERROR [backend 5/8] RUN conda env create -f /opt/environment.yml      34.6s

3.) Running the conda install command from the docker image on my Mac yields a similar error:

conda env create -f <snip>/matcher/backend/environment.yml 
Collecting package metadata (repodata.json): done
Solving environment: done
Downloading and Extracting Packages
libsqlite-3.42.0     | 804 KB    | ##################################### | 100% 
python-3.9.2         | 12.0 MB   | ##################################### | 100% 
libzlib-1.2.13       | 47 KB     | ##################################### | 100% 
setuptools-67.7.2    | 569 KB    | ##################################### | 100% 
sqlite-3.42.0        | 794 KB    | ##################################### | 100% 
pip-21.0.1           | 1.1 MB    | ##################################### | 100% 
ncurses-6.4          | 780 KB    | ##################################### | 100% 
openssl-1.1.1u       | 1.6 MB    | ##################################### | 100% 
zlib-1.2.13          | 78 KB     | ##################################### | 100% 
libcxx-16.0.5        | 1.1 MB    | ##################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done

Installing pip dependencies: | Ran pip subprocess with arguments:
['/opt/homebrew/anaconda3/envs/matcher-api/bin/python', '-m', 'pip', 'install', '-U', '-r', '<snip>/matcher/backend/condaenv.nf9w2agt.requirements.txt']

Pip subprocess output:
Collecting asyncpg==0.24.0
  Downloading asyncpg-0.24.0.tar.gz (787 kB)
Collecting fastapi==0.73.0
  Downloading fastapi-0.73.0-py3-none-any.whl (52 kB)
Collecting orjson==3.6.0
  Downloading orjson-3.6.0-cp39-cp39-macosx_10_9_x86_64.macosx_11_0_arm64.macosx_10_9_universal2.whl (436 kB)
Collecting pandas==1.4.1
  Downloading pandas-1.4.1-cp39-cp39-macosx_11_0_arm64.whl (10.5 MB)
Collecting pillow==8.3.1
  Downloading Pillow-8.3.1-cp39-cp39-macosx_11_0_arm64.whl (2.7 MB)
Collecting python-multipart==0.0.5
  Downloading python-multipart-0.0.5.tar.gz (32 kB)
Pip subprocess error:
ERROR: Could not find a version that satisfies the requirement rdkit-pypi==2021.3.5
ERROR: No matching distribution found for rdkit-pypi==2021.3.5
failed
CondaEnvException: Pip failed

**Comment by hooveran** [comment]
Hi @mccarren,

Incompatibility with ARM-based M1/M2 processors is a known issue, which is why we put an x86 processor in the [system requirements](https://github.com/Merck/Matcher#system-requirements-) .

As you point out, i think the source of the issue is the pinned rdkit version. I think a later version of rdkit is needed for ARM processors, but later rdkit versions become incompatible with the current postgres DB cartridge rdkit version, because these different rdkit versions have incompatible mol binaries. We query our DB with rdkit mol binaries, which lets us get away with transmitting crazy looking fragments like cleaved aromatic rings, in a way that isn’t nearly as robust were we to use SMILES or molfile serialization.

In summary, to support ARM processors, and not just x86 processors, we need to update both the backend and DB rdkit versions simultaneously. I’d welcome a PR if anyone would give it a shot, also possible we will do it in the future.

**Comment by mccarren** [comment]
Thank you so much for the explanation and quick response. Apologies that I read right over the clear requirements at the top, not sure how I missed that.  I'll mark this closed.  My coworker has it running on a Linux ec2 as a server (firewalled).  
Patrick
